Fix AT_USE_JITERATOR checks

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100464

`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

peterbell10 committed:
Fix AT_USE_JITERATOR checks

`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix AT_USE_JITERATOR checks"


`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix AT_USE_JITERATOR checks"


`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix AT_USE_JITERATOR checks"


`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix AT_USE_JITERATOR checks"


`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

[ghstack-poisoned]

peterbell10 commented on aten/src/ATen/native/cuda/UnaryGeometricAcosKernel.cu:
```diff
@@ -5,19 +5,21 @@
 namespace at::native {
 
-CONSTEXPR_EXCEPT_WIN_CUDA char acos_name[] = "acos";
+CONSTEXPR_EXCEPT_WIN_CUDA char acos_name[] = "acos_impl";
```
The `_impl` is necessary because otherwise it calls `acos` defined in the jiterator headers which fails because the template parameters are different. Same issue exists for all of these.

peterbell10 commented on aten/src/ATen/native/cuda/UnaryGeometricAcosKernel.cu:
```diff
@@ -5,19 +5,21 @@
   auto common_dtype = iter.common_dtype();
   if (at::isComplexType(common_dtype)) {
-#if AT_USE_JITERATOR
+    // Disabled due to accuracy issues
```
Some of these might just need tolerance bumps, but I didn't want to spend much time on this.

lezcano replied:
Could you open an issue describing what would need doing and tag it as "good first issue"?

peterbell10 replied:
Opened #100842

lezcano reviewed (approved):
Good catch!

Is there any way we could have a regression test for this (even if it's xfailed for now, given that all these are deactivated now).

Also cc @kshitij12345 as he probably wrote a few of these.

peterbell10 commented:
@pytorchbot rebase

peterbell10 committed:
Update on "Fix AT_USE_JITERATOR checks"


`AT_USE_JITERATOR` evaluates to false when the definition isn't
included, so these files were not using jiterator at all.

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge

ngimel reviewed (approved):
large/extreme values could have bugs in the implementation, libc++ might not handle those cases correctly (iirc we had to skip some mac tests for extreme values for the same reason)
